src/Castle.Windsor/MicroKernel/Context/CreationContext.cs

```diff
@@ -321,7 +321,11 @@ private bool CanConvertParameter(Type type)
 		private bool CanResolve(DependencyModel dependency, object inlineArgument)
 		{
 			var type = dependency.TargetItemType;
-			if (inlineArgument == null || type == null)
+			if (inlineArgument == null)
+			{
+				return type.IsNullable();
```

Comment:
This change appears fine, I can't think of anything this would cause a problem for. Since it is in a core area used by everything could you please add a unit test for the core functionality in addition to your typed factory one.